addons/pos_stripe/static/src/js/payment_stripe.js

```diff
@@ -0,0 +1,200 @@
+                    line.set_payment_status('retry');
+                    line.hideDeleteButton = false;
+                    return false;
+                } else {
+                    this.pos.connectedReader = this.payment_method.stripe_serial_number;
+                    return true;
+                }
+            }
+        }
+    },
+
+    collect_payment: async function (currency, amount) {
+        let line = this.pos.get_order().selected_paymentline;
+        let client_secret = await this.fetch_payment_intent_client_secret(currency, amount);
+        line.set_payment_status('waitingCard');
+        let collectPaymentMethod = await this.terminal.collectPaymentMethod(client_secret);
+        if (collectPaymentMethod.error) {
+            this._show_error(collectPaymentMethod.error.message, collectPaymentMethod.error.code);
+            line.set_payment_status('retry');
+            line.hideDeleteButton = false;
+            return false;
+        } else {
+            line.set_payment_status('waiting');
+            let processPayment = await this.terminal.processPayment(collectPaymentMethod.paymentIntent);
+            if (processPayment.error) {
+                this._show_error(processPayment.error.message, processPayment.error.code);
+                line.set_payment_status('retry');
+                line.hideDeleteButton = false;
+                return false;
+            } else if (processPayment.paymentIntent) {
+                line.set_payment_status('waiting');
+                let capture_payment = await this.capture_payment(processPayment.paymentIntent.id);
+                line.set_payment_status('done');
+                line.transaction_id = (capture_payment.id);
+                return true;
+            }
+        }
+    },
+
+    capture_payment: function (paymentIntentId) {
+        return rpc.query({
+            model: 'pos.payment.method',
+            method: 'stripe_capture_payment',
+            args: [paymentIntentId],
+        }, {
+            silent: true,
+        }).catch(function (error) {
+            this._show_error(_t('error'));
+        });
+    },
+
+    fetch_payment_intent_client_secret: function (currency, amount) {
+        return rpc.query({
+            model: 'pos.payment.method',
+            method: 'stripe_payment_intent',
+            args: [currency, amount],
+        }, {
+            silent: true,
+        }).catch(function (error) {
+            this._show_error(_t('error'));
+        }).then(function (data) {
+            return data.client_secret;
+        });
+    },
+
+    send_payment_request: async function (cid) {
+        await this._super.apply(this, arguments);
+        let line = this.pos.get_order().selected_paymentline;
+        line.hideDeleteButton = true;
+        line.set_payment_status('waiting');
+        return await this.check_reader() ? await this.collect_payment(this.pos.currency.name, line.amount) : false;
```

Comment:
```suggestion
        return await this.check_reader() ? await this.collect_payment(this.pos.currency.name, line.amount).catch(() => return false) : false;
```

What do you think about this? This is to ensure that if an error is thrown in the call to `collect_payment`, it will still resolve to `false`, which will be received by `_sendPaymentRequest`, so that it can set the proper state of the payment.